pkg/blockchain/plugin.go

```diff
@@ -129,23 +126,45 @@ type BatchPin struct {
 	// - The context is a function of:
 	//   - A single topic declared in a message - topics are just a string representing a sequence of events that must be processed in order
 	//   - A ledger - everone with access to this ledger will see these hashes (Fabric channel, Ethereum chain, EEA privacy group, Corda linear ID)
-	//   - A restricted group - if the mesage is private, these are the nodes that are elible to receive a copy of the private message+data
+	//   - A restricted group - if the mesage is private, these are the nodes that are eligible to receive a copy of the private message+data
 	// - Each message might choose to include multiple topics (and hence attach to multiple contexts)
 	//   - This allows multiple contexts to merge - very important in multi-party data matching scenarios
 	// - A batch contains many messages, each with one or more topics
 	//   - The array of sequence hashes will cover every unique context within the batch
 	// - For private group communications, the hash is augmented as follow:
 	//   - The hashes are salted with a UUID that is only passed off chain (the UUID of the Group).
 	//   - The hashes are made unique to the sender
-	//   - The hashes contain a sender specific nonce that is a monotomically increasing number
+	//   - The hashes contain a sender specific nonce that is a monotonically increasing number
 	//     for batches sent by that sender, within the context (maintined by the sender FireFly node)
 	Contexts []*fftypes.Bytes32
+
+	// Event contains info on the underlying blockchain event for this batch pin
+	Event Event
+}
+
+type Event struct {
```

Comment:
~I think in our current codebase, this object might be better located at `fftypes.BlockchainEvent`~ understand the difference now